### PR TITLE
feat: add new typography and spacing values

### PR DESCRIPTION
**Pull request by melaniebmn**
# Summary | Résumé

Updating our typography and spacing values based on the new values we decided on in alignment with Canada.ca standards.

## Updates applied in this PR:

- Updated typography core values to adjust our vertical rhythm calculations.
- New font size values for all desktop & mobile font size tokens.
- New line-height values for all desktop & mobile line-height tokens.
- New spacing values for all existing spacing tokens.
- Renamed `caption` to `text-small`.
- Added global font token for `text-mono`.
- Addition of new spacing tokens for increased flexibility.
- Mapping of all old spacing tokens to the matching new spacing token for component tokens.
- The font tokens for the text-long tokens have been removed, as all text tokens will be using the same line-height moving forward.

## Documents

The document for the new typography and spacing decisions can be found [here](https://docs.google.com/document/d/1duGEYoerJaYj_6oGKFGJtLv6PaOS0SUcRlftNbe33iA/edit?tab=t.0).

## Fixes

I noticed that the compiled text tokens file was also loading in textarea tokens, and the font tokens file was also loading in font sizes + weights. I added some fixes to the config to avoid accidentally including tokens from sibling directories starting with the same name.

NOTE: DO NOT MERGE YET! PRE-RELEASE MESSAGING NEEDS TO GO OUT FIRST!

### CODE REVIEW

File: tokens/global/spacing/tokens.js [hunk 1]
```diff
@@ -1,72 +1,20 @@
...
-  },
-};
+// Calculate constant multiplier once to avoid redundant calculations in loop
+const multiplier = base.fontSize.value * base.lineHeight.value / 400;
+
+/**
+  * Loop starting with a 25 increment to provide more options for smaller
+  * spacing values. Switches to a 50 increment after reaching 250.
+ */
+for (let i = 0; i <= 1250; i += i >= 250 ? 50 : 25) {
```
Comment (daine) [inline tokens/global/spacing/tokens.js:13]: 🤩

**Review by daine — COMMENTED** [review]
I did a quick pass but I'll come back to it on Monday to help check for other things too. It's looking really good so far!

File: tokens/global/typography/lineHeights.js [hunk 2]
```diff
@@ -1,56 +1,64 @@
...
-  const fontSizeValue = parseFloat(fontSize) / base.fontSize.value;
-  const numBaselines =
-    Math.ceil((fontSizeValue * octave) / base.lineHeight.value) / octave +
-    1 / octave;
+// Set baseline based on typography core values
+const baseline = rootFontSize * baseFontSize * baseLineHeight;
 
-  return (numBaselines * base.lineHeight.value * 100) / fontSizeValue;
+// Multiplier definitions for desktop and mobile
+const multipliers = {
```
[the remark this answers is not in the document]
Comment (daine) [reply tokens/global/typography/lineHeights.js:13]: This new lookup works pretty well!

**Review by daine — DISMISSED** [review]
The rem values actually have less decimal points which was a good surprise to get the equivalent pixels! I tested this on gcds-components and the docs site, approving it!

**Review by ethanWallace — DISMISSED** [review]
LGTM!

**Comment by melaniebmn** [comment]
NOTE: DO NOT MERGE YET! PRE-RELEASE MESSAGING NEEDS TO GO OUT FIRST!

**Comment by melaniebmn** [comment]
@daine @ethanWallace I added the token updates needed after the typography and spacing updates.

File: tokens/global/typography/fontSizes.js [hunk 3]
```diff
@@ -14,10 +14,20 @@ const headingsMultiplier = {
...
+  * convert back to rem by dividing by 16.
+  */
+  const roundedFontSize = Math.round(remFontSize * rootFontSize.value) / rootFontSize.value;
 
   return {
-    value: `${fontSizeValue}rem`,
+    value: `${roundedFontSize}rem`,
     type: 'fontSizes',
   };
 };
```
Comment (daine) [inline tokens/global/typography/fontSizes.js:33]: Did we say we were going to rename the `caption` token to `small`?

Comment (melaniebmn) [reply tokens/global/typography/fontSizes.js:33]: Let's check in the meeting today - it would be another breaking change.

Comment (melaniebmn) [reply tokens/global/typography/fontSizes.js:33]: @daine the token has been renamed.

**Review by daine — DISMISSED** [review]
🚀 ship it!

**Review by daine — APPROVED** [review]
🚀 🚀

**Review by ethanWallace — APPROVED** [review]
LGTM!

**Pull request merged**